src/frontend/utils/utilityTypes.ts

```diff
@@ -0,0 +1,13 @@
+
+/**
+ * PartialRecordWithValues
+ * Allows creating a type based on mapped type K that holds a value o type V (`string` by default)
+ * That makes all keys from mapped type optional but requires so that they have value.
+ * { } // ok
+ * { x: 1, y: 2 } // ok
+ * { x: 1 } // ok
+ * { x: 1, y: undefined } // error
```

Comment:
please add `@private` jsdoc tag (so it won't be present in the generated documentation) and remove the extra line (11)